pallets/settlement/src/lib.rs

```diff
@@ -1244,6 +1285,18 @@ impl<T: Trait> Module<T> {
         ))
     }
 
+    // Get the public MERCAT account and the pending on-chain balance.
+    fn get_pending_state(
+        did: &IdentityId,
+        account_id: &MercatAccountId,
+        instruction_id: u64,
+    ) -> Result<EncryptedAmount, DispatchError> {
+        Ok(
+            <ConfidentialAsset<T>>::mercat_tx_pending_state((did, account_id, instruction_id))
+                .to_mercat::<T>()?,
+        )
```

Comment:
```suggestion
        <ConfidentialAsset<T>>::mercat_tx_pending_state((did, account_id, instruction_id))
                .to_mercat::<T>()
```
(unless `?` is doing a necessary type conversion here)